pkg/ipam/node.go

```diff
@@ -598,6 +601,26 @@ func (n *Node) maintainIPPool(ctx context.Context) error {
 
 	// Release excess addresses
 	if a.release != nil && len(a.release.IPsToRelease) > 0 {
+		var ipsToRelease []string
+		// Since, the default CRD sync interval is 15 secs, there might be new allocations that aren't reflected in
+		// the ciliumnode object yet. Verify against pod cache to check if any of the IPs are in use.
+		for _, ip := range a.release.IPsToRelease {
+			if watchers.PodStore == nil {
+				scopedLog.Warnf("Pod store un-initialized")
+				continue
+			}
+			values, err := watchers.PodStore.(cache.Indexer).ByIndex(watchers.PodIpIndex, ip)
+			if err != nil {
+				scopedLog.WithError(err).Error("Unable to access pod store index")
+				continue
+			}
+			if len(values) != 0 {
```

Comment:
Do we risk leaking IPs for `Completed` (and maybe Evicted) pods?
I think a direct NewIndexInformer would handle that (but am not sure!).

Reply:
That's a great question. Will look into the possibility. Maybe i should check the status of pods as well.

Reply:
Just added some changes to handle `Failed` and `Succeeded` phases. https://github.com/DataDog/cilium/pull/20/files#diff-01e105c00ecf377284f9b2f24279d5ac0ff6c96a8bc076265905ab3e982ff384R637